### PR TITLE
[DOCS] SQL: Document async SQL search

### PR DESCRIPTION
**Pull request by jrodewig**
Documents async SQL search functionality.

I plan to add formal API documentation for the async APIs with a later PR.

Relates to #73991 and #74845.

### Previews
* Run an async SQL search: https://elasticsearch_74975.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.14/sql-async.html
* Supported REST parameters: https://elasticsearch_74975.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.14/sql-rest-fields.html

### CODE REVIEW

**Comment by jrodewig** [comment]
Rebased this against 7.14 to include the release highlight.

File: docs/reference/sql/endpoints/rest.asciidoc [hunk 1]
```diff
@@ -561,6 +562,170 @@ Frank Herbert  |Dune           |604            |1965-06-01T00:00:00.000Z|TUESDAY
...
+  "expiration_time_in_millis": 1611690295000,
+  "completion_status": 200
+}
+----
+// TESTRESPONSE[s/FnR0TDhyWUVmUmVtWXRWZER4MXZiNFEad2F5UDk2ZVdTVHV1S0xDUy00SklUdzozMTU=/$body.id/]
+// TESTRESPONSE[s/"expiration_time_in_millis": 1611690295000/"expiration_time_in_millis": $body.expiration_time_in_millis/]
+
+To get the results, use the search ID with the get async SQL search API. If the
+search is still running, specify how long you’d like to wait using
+`wait_for_completion_timeout`. You can also specify the response `format`.
```
Comment (bpintea) [inline docs/reference/sql/endpoints/rest.asciidoc:638]: 👍

File: docs/reference/sql/endpoints/rest.asciidoc [hunk 2]
```diff
@@ -561,6 +562,170 @@ Frank Herbert  |Dune           |604            |1965-06-01T00:00:00.000Z|TUESDAY
...
+  "keep_alive": "2d",
+  "wait_for_completion_timeout": "2s",
+  "query": "SELECT * FROM library ORDER BY page_count DESC",
+  "fetch_size": 5
+}
+----
+// TEST[setup:library]
+
+You can use the get async SQL search API's `keep_alive` parameter to later
+change the retention period. The new period starts after the request runs.
```
Comment (bpintea) [inline docs/reference/sql/endpoints/rest.asciidoc:667]: Not sure if it's worth mentioning that `wait_for_completion_timeout` and `keep_alive` can be used concurrently, given that we don't have an enumeration of `_sql/async` supported URL params, _or_ if we generally leave it to the users to intuit that.

**Review by bpintea — APPROVED** [review]
Not sure why the docs preview isn't based off this patch, but the changes LGTM.

**Comment by bpintea** [comment]
> I plan to add formal API documentation for the async APIs with a later PR.

Many thanks for this PR and for the API documentation, @jrodewig.
CC @stevejgordon

File: docs/reference/sql/endpoints/rest.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/sql/endpoints/rest.asciidoc:667]: Good point. I'll add `wait_for_completion_timeout` to the example snippet.

**Pull request merged**